apps/src/lib/kits/maker/Button.js

```diff
@@ -19,4 +19,25 @@ export default function Button(opts) {
     get: () => this.value === 1
   });
 }
-Button.inherits(five.Button);
+PlaygroundButton.inherits(five.Button);
+
+export function MicroBitButton(board) {
+  // There are six button events, ['', 'down', 'up', 'click', 'long-click', 'hold']
+  this.buttonEvents = new Array(6).fill(0);
+  board.mb.addFirmataEventListener((sourceID, eventID) => {
+    if (board.pin === sourceID) {
+      this.buttonEvents[eventID]++;
+      if (eventID === 1) {
+        this.emit('down');
+      } else if (eventID === 2) {
+        this.emit('up');
+      }
+    }
+  });
+
+  // Add a read-only `isPressed` property
+  Object.defineProperty(this, 'isPressed', {
+    get: () => this.buttonEvents[1] > 0
```

Comment:
Why index 1?

Reply:
Index 1 is the 'button went down' event.

Reply:
Sorry might be missing something obvious but: Wouldn't this return true always after the first time the button is pressed, not just when the button is *currently* pressed?

Reply:
Good catch - updated!